### PR TITLE
[FEATURE] Add basic Auditing logger

### PR DESCRIPTION
**Pull request by dfeyer**
This change introduce an auditing log. The current version store
informations related to the Command and their respective CommandHandler.

### CODE REVIEW

File: Classes/Etg24/EventSourcing/Auditing/CommandSerializerInterface.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (Akii) [inline Classes/Etg24/EventSourcing/Auditing/CommandSerializerInterface.php]: We already have a MessageSerializer afaik, need to use that one.

File: Classes/Etg24/EventSourcing/Command/Bus/InternalCommandBus.php [hunk 2]
```diff
@@ -41,10 +42,17 @@ public function handle(Command $command) {
 		$commandHandled = FALSE;
 
 		foreach ($this->commandHandlers as $commandHandler) {
+			/** @var CommandHandler $commandHandler */
```
Comment (Akii) [inline Classes/Etg24/EventSourcing/Command/Bus/InternalCommandBus.php]: PhpStorm should detect the type because `@var CommandHandler[]`

**Comment by Akii** [comment]
I like the approach so far, will review it more this evening.

File: Classes/Etg24/EventSourcing/Auditing/CommandLogger.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+use Etg24\EventSourcing\Command\Handler\CommandHandler;
+use TYPO3\Flow\Annotations as Flow;
+
+/**
+ * Command Auditing
+ */
+class CommandLogger {
+
+	/**
+	 * @Flow\Inject
```
Comment (Akii) [inline Classes/Etg24/EventSourcing/Auditing/CommandLogger.php]: Another thing, for consistency sake, @var must go before annotations. I know this differs from the TYPO3 CGL, but that's just how it is.

**Comment by Akii** [comment]
Something similar should be available for event handling.

**Comment by dfeyer** [comment]
I agree, this is the next step, I'm reading more documentation on how to handle correctly the audit log before implementation.

**Comment by dfeyer** [comment]
I fixed your comments and add a basic Event auditing object. Ready for review, maybe we can avoid some code duplication between CommandLogger and EventLogger, but not sure if it's a really good idea

**Comment by dfeyer** [comment]
For the future the Auditing can be improved by attaching Event log the the command that trigger the events, like this the log can be more easy to read (currently the Event are logged before the Command success/failure, code wise it's right, but for the user, can be hard to read.)

**Comment by svparijs** [comment]
@dfeyer you commit is merged in a fork, which will be further developed. https://github.com/Flow2Lab/EventSourcing

**Pull request closed without merging**